File: Algo/Algorithm/Questions/Arrays/09.isMonotonic/main.js
// O(n) time | O(1) space - where n is the length of the array
function isMonotonic(array) {
  let isNonIncreaseaing = true;
  let isNonDecreaseaing = true;

  for (let i = 1; i < array.length; i += 1) {
    if (array[i] > array[i - 1]) {
      isNonDecreaseaing = false;
    }

    if (array[i] < array[i - 1]) {
      isNonIncreaseaing = false;
    }
  }

  return isNonDecreaseaing || isNonIncreaseaing;
}

File: Algo/Algorithm/Questions/Arrays/09.isMonotonic/main.cpp
// O(n) time | O(1) space - where n is the length of the array.
#include <vector>
#include <iostream>
using namespace std;

bool isMonotonic(vector<int> array)
{
  if (array.size() <= 2)
  {
    return true;
  }

  for (int i = 0; i < array.size() - 1; i++)
  {
    if (abs(array[i]) > abs(array[i + 1]))
      return false;

    if (array[i] * array[i + 1] < 0)
      return false;
  }

  return true;
}

int main()
{
  vector<int> array = {-1, -5, 10};
  cout << isMonotonic(array);
  cout << endl;
}
